test/meta.cpp

```diff
@@ -743,6 +761,15 @@ TEST_F(Meta, MetaDtor) {
     ASSERT_EQ(empty_type::counter, 0);
     ASSERT_TRUE(dtor.invoke(empty));
     ASSERT_EQ(empty_type::counter, 1);
+
+    dtor = meta::resolve<empty_type*>().dtor();
```

Comment:
>I think we should handle it as as special case and use a pointer to a member pointer, which should be the same length as void* then.

In fact, you must exclude both pointers to members and pointers to functions, the latter cannot be converted to `void *` as well.  
However, I'd flip the problem on its head and check that `T` is a pointer and `std::remove_pointer_t<T>` isn't a function. `std::is_pointer_v` is already false for members.  
After all, what you want is to treat pointers to classes transparently. In this case, you can use a `static_cast` rather than a `reinterpret_cast`, so that the compiler is allowed to tell you that you're not doing a wrong conversion. In all the other cases, `return ptr;` works just fine and you don't need casts at all.